### PR TITLE
Refactor files

### PR DESCRIPTION
**Pull request by loicgasser**
With this PR it is now very simple to apply the same logic used for the KMLs to any kind of file.

These changes can be applied in master as there are no functional changes.

The actual service to save the styles will be created on a separated branch called `mvt`.

[Test Link](https://mf-geoadmin3.dev.bgdi.ch/?api_url=%2F%2Fmf-chsdi3.dev.bgdi.ch%2Frefactor_files)

### CODE REVIEW

File: chsdi/lib/files_handler.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            content_encoding = 'gzip'
+            data = gzip_string(data)
+        if self.admin_id is not None:
+            status = 'updated'
+        else:
+            # In case the file already exist, we create a fork
+            status = 'copied'
+            self._fork()
+        forked = status == 'copied'
+        self.s3_fileshanlder.save_object(self.file_path, mime, content_encoding, data, not forked)
```
Comment (procrastinatio) [inline chsdi/lib/files_handler.py]: typo s3_fileshanlder --> s3_fileshandler

File: chsdi/lib/files_handler.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+from chsdi.lib.helpers import gzip_string
+from chsdi.models.clientdata_dynamodb import get_dynamodb_table, get_bucket
+
+
+class DynamoDBFilesHandler:
+
+    def __init__(self, table_name, bucket_name):
+        # We use instance roles
+        self.table = get_dynamodb_table(table_name=table_name)
```
Comment (procrastinatio) [inline chsdi/lib/files_handler.py]: why not pass the dynamo table directly?

Comment (loicgasser) [reply chsdi/lib/files_handler.py]: To be consistent with S3FilesHandler.

File: chsdi/lib/files_handler.py (hunk 1 above)
Comment (loicgasser) [reply chsdi/lib/files_handler.py]: Good catch

**Comment by loicgasser** [comment]
@procrastinatio typo fixed thanks for the review

**Comment by loicgasser** [comment]
@pauloamado done in last commit.
image
Let me know if you figure out the permalink for the lang.

**Comment by rebert** [comment]
@loicgasser thanks for fixing the test :-)

**Comment by loicgasser** [comment]
@danduk82 this one is for master

**Pull request merged**